### PR TITLE
update vis-network

### PR DESCRIPTION
**Pull request by antgonza**

### CODE REVIEW

**Comment by antgonza** [comment]
@justinshaffer, I deployed this version in qiita-rc and it will be great if you could tests that everything seems fine (running the tutorials should suffice). Thank you!

**Comment by antgonza** [comment]
@ElDeveloper, thank you for the review. Note that during @justinshaffer review in qiita-rc, he found that there is some kind of bug while using the new version and we are evaluating if we want to do try to have this PR for this release or next.

Now, answering your question:
- sortMethod: "directed": completely changes behavior in the new version and is better to use the default setting.
- node-size/levelSeparation: it's just to save space (shrink separation between nodes)

**Comment by ElDeveloper** [comment]
Makes a lot of sense!

**Comment by antgonza** [comment]
@ElDeveloper, this version is deployed in qiita-rc and @justinshaffer and I have tested and seems to be fine; could you do another review? Thank you!

File: qiita_pet/static/js/networkVue.js [hunk 1]
```diff
@@ -713,22 +713,12 @@ Vue.component('processing-graph', {
...
-      $(job_info.outputs).each(function(){
-        var out_name = this[0];
-        var out_type = this[1];
-        var n_id = job_info.id + ":" + out_name;
-        vm.edges_ds.add({id: vm.edges_ds.length + 1, from: job_info.id, to: n_id });
-        vm.nodes_ds.add({id: n_id, shape: 'dot', label: formatNodeLabel(out_name + "\n(" + out_type + ")"), group: "type", name: out_name, type: out_type});
-      });
-      vm.network.redraw();
+
+      vm.new_job_info = {
```
Comment (ElDeveloper) [inline qiita_pet/static/js/networkVue.js]: Am I reading this right, do we no longer need to manually add edges and nodes?

File: qiita_pet/static/js/networkVue.js [hunk 2]
```diff
@@ -713,22 +713,12 @@ Vue.component('processing-graph', {
      **/
     addJobNodeToGraph: function(job_info) {
       let vm = this;
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/static/js/networkVue.js]: We do as we need to add vm.new_job_info below.

File: qiita_pet/static/js/networkVue.js (hunk 1 above)
Comment (antgonza) [reply qiita_pet/static/js/networkVue.js]: That is correct, now we reload the network (vm.updateGraph();) this assures that we get the final position of the new nodes and simplifies the code.

File: qiita_pet/static/js/networkVue.js [hunk 3]
```diff
@@ -974,11 +977,16 @@ Vue.component('processing-graph', {
           // data in a way that Vis.Network likes it
           // Format edge list data
           for(var i = 0; i < data.edges.length; i++) {
+            // forcing a string
+            data.edges[i][0] =data.edges[i][0].toString()
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/static/js/networkVue.js]: thanks

File: qiita_pet/static/js/networkVue.js [hunk 4]
```diff
@@ -974,11 +977,16 @@ Vue.component('processing-graph', {
...
+            data.edges[i][1] =data.edges[i][1].toString()
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/static/js/networkVue.js]: thanks.

File: qiita_pet/templates/sitebase.html [hunk 5]
```diff
@@ -20,6 +20,8 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en-US">
   <head>
     <title>{{portal_styling.title}}</title>
+    <!-- https://unpkg.com/browse/vis-network@6.5.2/styles/vis-network.min.css -->
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/templates/sitebase.html]: I left there so we know where that file came from, should I added somewhere else; like the README?

File: qiita_pet/templates/sitebase.html [hunk 6]
```diff
@@ -45,7 +45,8 @@
     <script src="{% raw qiita_config.portal_dir %}/static/vendor/js/jquery.dataTables.plugin.natural.js" type="text/javascript"></script>
     <script src="{% raw qiita_config.portal_dir %}/static/vendor/js/vue.min.js"></script>
     <script src="{% raw qiita_config.portal_dir %}/static/js/qiita.js"></script>
-    <script src="{% raw qiita_config.portal_dir %}/static/vendor/js/vis.min.js"></script>
+    <!-- https://unpkg.com/browse/vis-network@6.5.2/standalone/umd/vis-network.min.js -->
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/templates/sitebase.html]: Same that before; know where it comes from.

File: qiita_pet/templates/study_base.html [hunk 7]
```diff
@@ -4,7 +4,7 @@
 {% set study_title_msg = study_title.replace('"', '\\"') %}
 <link rel="stylesheet" src="{% raw qiita_config.portal_dir %}/static/vendor/css/vis.min.css" type="text/css">
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/templates/study_base.html]: AFAIK same; in fact this file didn't change ...

File: qiita_pet/templates/sitebase.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (ElDeveloper) [reply qiita_pet/templates/sitebase.html]: I would suggest putting these links in the License file for this dependency. I think there should be a markdown file in the vendor directory.

File: qiita_pet/static/js/networkVue.js (hunk 1 above)
Comment (ElDeveloper) [reply qiita_pet/static/js/networkVue.js]: Got it, thanks!

File: qiita_pet/static/vendor/css/vis-network.min.css [hunk 8]
```diff
@@ -1 +1,971 @@
...
\ No newline at end of file
+<!DOCTYPE html><html lang="en"><head><script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-140352188-1"></script><script>window.dataLayer = window.dataLayer || [];
```
[the remark this answers is not in the document]
Comment (antgonza) [reply qiita_pet/static/vendor/css/vis-network.min.css]: Note that this is the minified version available online:
```shell
(qiita) 06:48:52 qiita@update-vis.js$ git status
On branch update-vis.js
nothing to commit, working tree clean
(qiita) 06:48:53 qiita@update-vis.js$ curl -o vis-network.min.css "https://unpkg.com/browse/vis-network@6.5.2/styles/vis-network.min.css"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  558k    0  558k    0     0  4243k      0 --:--:-- --:--:-- --:--:-- 4263k
(qiita) 06:48:58 qiita@update-vis.js$ md5 vis-network.min.css qiita_pet/static/vendor/css/vis-network.min.css 
MD5 (vis-network.min.css) = 6fe5886e64b1515ad1cfbb17ff6fc630
MD5 (qiita_pet/static/vendor/css/vis-network.min.css) = 6fe5886e64b1515ad1cfbb17ff6fc630
```

[the remark this answers is not in the document]
Comment (ElDeveloper) [reply qiita_pet/static/vendor/css/vis-network.min.css]: Alright, looks good to me then.

**Pull request merged**